dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinUserMetadataRestController.java

```diff
@@ -303,19 +308,87 @@ private void sendEmailWithDownloadLink(Context context, DSpaceObject dso,
             bean.addRecipient(email);
             bean.send();
         } catch (MessagingException e) {
-            log.error("Cannot send the email because: " + e.getMessage());
+            log.error("Cannot send the email with download link, because: " + e.getMessage());
+            throw new MessagingException(e.getMessage());
+        }
+        // If previous mail fails with exception, this block never executes = admin is NOT
+        // notified, if the mail is not really sent (if it fails HERE, not later, e.g. due to mail server issue).
+        sendAdminNotificationEmail(context, downloadLink, dso, clarinLicense, mailType, clarinUserMetadataRestList);
+
+    }
+
+    private void sendAdminNotificationEmail(Context context,
+                                            String downloadLink,
+                                            DSpaceObject dso,
+                                            ClarinLicense clarinLicense,
+                                            MailType mailType,
+                                            List<ClarinUserMetadataRest> extraMetadata)
+            throws MessagingException, IOException {
+        try {
+            Locale locale = context.getCurrentLocale();
+            Email email2Admin = Email.getEmail(I18nUtil.getEmailFilename(locale, "clarin_download_link_admin"));
+            String ccAdmin = configurationService.getProperty("download.email.cc");
+            List<String> ccEmails = new ArrayList<>();
+            if (ccAdmin != null && !ccAdmin.isEmpty()) {
+                ccEmails.add(ccAdmin);
+            }
+            String licenseName = clarinLicense.getName().trim().replace(" ", "_").toLowerCase();
+            String[] licenseSpecialRecipients = configurationService.getArrayProperty("download.email.cc."
+                    + licenseName);
+            if (licenseSpecialRecipients != null) {
+                for (String cc : licenseSpecialRecipients) {
+                    if (!cc.isEmpty()) {
+                        ccEmails.add(cc.trim());
+                    }
+                }
+            }
+
+            if (!ccEmails.isEmpty()) {
+                for (String cc : ccEmails) {
```

Comment:
private void addEmailArguments(Email email, MailType mailType, DSpaceObject dso, String downloadLink,
                               ClarinLicense clarinLicense, Context context, List<ClarinUserMetadataRest> extraMetadata)
        throws MessagingException {
    switch (mailType) {
        case ALLZIP:
            email.addArgument("all files requested");
            break;
        case BITSTREAM:
            email.addArgument(dso.getName());
            break;
        default:
            throw new IllegalArgumentException("The mail type is not supported.");
    }

    email.addArgument(downloadLink);
    email.addArgument(clarinLicense.getDefinition());

    if (context.getCurrentUser() != null) {
        email.addArgument(context.getCurrentUser().getFullName());
        email.addArgument(context.getCurrentUser().getEmail());
    } else {
        email.addArgument("Anonymous user");
        email.addArgument("Anonymous user");
    }

    StringBuilder exdata = new StringBuilder();
    for (ClarinUserMetadataRest data : extraMetadata) {
        exdata.append(data.getMetadataKey()).append(": ").append(data.getMetadataValue()).append(", ");
    }
    email.addArgument(exdata.toString());
}

Reply:
I just renamed the function to "addAdminEmailArguments", since it would not work with another email. It is not a generic method for adding arguments to email, as your name suggests.